### ISSUE TITLE
 Refactor/reorganize agent initialization and manage actions

### ISSUE THREAD

**Issue by todor-ivanov** [dmwm/WMCore#11627]
**Impact of the new feature**
WMAgent

**Is your feature request related to a problem? Please describe.**
Yes. This issue succeeds this one: https://github.com/dmwm/WMCore/issues/8797.

We are now moving towards running the WMAgent inside a Docker container.  In this process we have introduced a new `run.sh` script which drives the agent startup and initialization inside the Docker container. Previously the whole process was driven only by the `manage` script, and now we started to face some duplicate mechanisms in keeping track of all steps followed. Namely:
* We create a subset of `.init` files populated by the `manage` script which serve to keep track the steps needed to be triggered only once at the beginning of an agent's lifetime (e.g. `$INSTALL_MYSQL/.init`). 
* At the same time we have implemented the equivalent mechanism for a set of `.dockerInit`  files populated by the `run.sh` script and serving the same purpose but from the container perspective. 
 
The more we go in that direction the closer to a locking condition we get. Since we are now about to consider all agent actions related to the Docker container management as well, we should start thinking of  a way to unite those mechanism in a single place and avoid future complications.  

There are different scenarios we need to cover. Some of them involve database refresh or complete deletion, some of them not :
* Agent starts for the first time - all databases need to be populated &&  agent configuratino needs to be generated out of `WMAgent.secrets` file
* Agent normal restart -  nothing needs to be 
* Agent being reconfigured through `WMAgent.secrets` - 
* Agent being reconfigured through agent config file
* Agent being patched and reconfigured 
* Agent's resource control needs to be updated  
* Eventual Agent complete cleanup - both databases and job areas etc.

Here to be filled in the int flow diagram:
```
...
```   

**Describe the solution you'd like**
Split the initialization and manage actions to be driven by different scripts. 
* initialization steps by `run.sh` 
* manage actions by `manage.sh`

The former should deal with all `.init` files and keeping track of the correct agent initialization flow.
The later should deal only with manage actions (e.g. talking to databases, populating tables, calling resorce control etc.), which should be called from `run.sh`. The communication between them both should happen only based on return codes, and the handling of errors needs to happen in `run.sh` and to be in sync with the established init sequence.
   
**Describe alternatives you've considered**
None

**Additional context**
Depends on:
* https://github.com/dmwm/WMCore/issues/8797
* https://github.com/dmwm/CMSKubernetes/pull/1364

Part of the following meta issue: https://github.com/dmwm/WMCore/issues/11314

**Comment by vkuznet** [comment]
I can take this ticket if @todor-ivanov or @amaltaro will not object. Please let me know.

**Comment by amaltaro** [comment]
@vkuznet , I believe it might be better to leave this one to @todor-ivanov , given that he spent many weeks working on the initial steps to deploying PyPi WMAgent in a docker container and being able to start services up. But I am fine with whatever you two decide to do.

**Comment by vkuznet** [comment]
that's fine with me if @todor-ivanov will continue this effort. I'm walking though list of possible High priority ticket which were unassigned.

**Comment by todor-ivanov** [comment]
Thanks @vkuznet, I am definitely going to continue on that one. This issue is a natural continuation on the basic WMAgent image creation, during which I had to struggle a lot with multiple workarounds in the agent initialization process for keeping backwards compatibility etc. Many of those we have widely discussed with @amaltaro. I have marked few very crucial points for fixing with this issue and I already have some concrete ideas, which I am eager to implement.
